### PR TITLE
smpeg2: use github artifact

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/smpeg2.rb [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+  url "https://github.com/icculus/smpeg/archive/refs/tags/release_2_0_0.tar.gz"
+  sha256 "fdd431bd607efcf0f35789fb3105d7535d4f0e8b46e673e9c0051726e8d1e701"
+  license "LGPL-2.1-or-later"
+  head "https://github.com/icculus/smpeg.git", branch: "main"
 
   livecheck do
-    url "https://svn.icculus.org/smpeg/tags/"
-    regex(%r{href=.*?release[._-]v?(2(?:[._]\d+)+)/}i)
+    url :stable
+    regex(/^release[._-]v?(\d+(?:[._]\d+)+)$/i)
```
Comment (samford) [inline Formula/s/smpeg2.rb:11]:
```suggestion
    regex(/^release[._-]v?(2(?:[._]\d+)+)$/i)
```

If this formula is only for 2.x versions, we should continue to restrict matching to 2.x tags.

Comment (chenrui333) [reply Formula/s/smpeg2.rb]: 👍

**Review by p-linnane — APPROVED** [review]

File: Formula/s/smpeg2.rb [hunk 2]
```diff
@@ -1,12 +1,19 @@
 class Smpeg2 < Formula
   desc "SDL MPEG Player Library"
   homepage "https://icculus.org/smpeg/"
-  url "svn://svn.icculus.org/smpeg/tags/release_2_0_0/", revision: "408"
-  head "svn://svn.icculus.org/smpeg/trunk"
+  # license change was done in 2021 Aug, which is 8 years after 2.0.0 release
+  # commit ref, https://github.com/icculus/smpeg/commit/ffa0d54
+  url "https://github.com/icculus/smpeg/archive/refs/tags/release_2_0_0.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/s/smpeg2.rb:6]: It says 8 years for both 0.4.5 and 2.0.0, are these both correct?

Also if so: why are we on an 8 year old version?

Comment (chenrui333) [reply Formula/s/smpeg2.rb:6]: mostly because it is buildable and running? (same goes to lots of other formulae)

Comment (chenrui333) [reply Formula/s/smpeg2.rb:6]: upstream does have some new release chat, https://github.com/icculus/smpeg/issues/9

Comment (MikeMcQuaid) [reply Formula/s/smpeg2.rb:6]: Yeh, that's fine, just checking it was correct here as comment same in both place.

**Review by MikeMcQuaid — APPROVED** [review]

File: Formula/s/smpeg2.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/s/smpeg2.rb:6]: haha, yeah, the 2.0.0 and 0.4.5 got released almost side to side.

**Comment by chenrui333** [comment]
rebased to fix the merge conflict with `audit_exceptions/permitted_formula_license_mismatches.json`

**Review by iMichka — APPROVED** [review]

**Pull request merged**